show/main.py

```diff
@@ -1170,6 +1171,91 @@ def line():
     """Show all /dev/ttyUSB lines and their info"""
     cmd = "consutil show"
     run_command(cmd, display_cmd=verbose)
+    # TODO: Stub
+    return
+
+
+@cli.group(cls=AliasedGroup, default_if_no_args=False)
+def warm_restart():
+    """Show warm restart configuration and state"""
+    pass
+
+@warm_restart.command()
+@click.option('-s', '--redis-unix-socket-path', help='unix socket path for redis connection')
+def state(redis_unix_socket_path):
+    """Show warm restart state"""
+    kwargs = {}
+    if redis_unix_socket_path:
+        kwargs['unix_socket_path'] = redis_unix_socket_path
+
+    data = {}
+    db = SonicV2Connector(host='127.0.0.1')
+    db.connect(db.STATE_DB, False)   # Make one attempt only
+
+    TABLE_NAME_SEPARATOR = '|'
+    prefix = 'WARM_RESTART_TABLE' + TABLE_NAME_SEPARATOR
+    _hash = '{}{}'.format(prefix, '*')
+    table_keys = db.keys(db.STATE_DB, _hash)
+
+    def remove_prefix(text, prefix):
+        if text.startswith(prefix):
+            return text[len(prefix):]
+        return text
+
+    table = []
+    for tk in table_keys:
+        entry = db.get_all(db.STATE_DB, tk)
+        r = []
+        r.append(remove_prefix(tk, prefix))
+        r.append(entry['restart_count'])
+
+        if 'state' not in  entry:
+            r.append("")
+        else:
+            r.append(entry['state'])
+
+        table.append(r)
+
+    header = ['name', 'restart_count', 'state']
+    click.echo(tabulate(table, header))
+
+@warm_restart.command()
+@click.option('-s', '--redis-unix-socket-path', help='unix socket path for redis connection')
+def config(redis_unix_socket_path):
+    """Show warm restart config"""
+    kwargs = {}
+    if redis_unix_socket_path:
+        kwargs['unix_socket_path'] = redis_unix_socket_path
+    config_db = ConfigDBConnector(**kwargs)
+    config_db.connect(wait_for_init=False)
+    data = config_db.get_table('WARM_RESTART')
+    keys = data.keys()
+
+    def tablelize(keys, data):
+        table = []
+
+        for k in keys:
+            r = []
+            r.append(k)
+
+            if 'enable' not in  data[k]:
+                r.append("false")
+            else:
+                r.append(data[k]['enable'])
+
+            if 'timer_name' in  data[k]:
+                r.append(data[k]['timer_name'])
+                r.append(data[k]['timer_duration'])
+            else:
+                r.append("NULL")
+                r.append("NULL")
+
+            table.append(r)
+
+        return table
+
+    header = ['name', 'enable', 'timer_name', 'timer_duration']
```

Comment:
header should be adjusted to the new schema.

Reply:
How do we present multiple timers at CLI?  Would it make more sense to have the real timer name at the header?

Of course, it is up to people who work on other timers to add the CLI later.

Reply:
I don't think putting fixed timer name in header is a good idea.
If there are multiple timers per docker, just fill space in other fields and make multiple rows. 
Anyway it is kind of UI things,  people may have different opinions.

Reply:
OK,  it is fine as is as baseline.